app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/display/DisplayFragmentTest.kt

```diff
@@ -93,4 +98,23 @@ class DisplayFragmentTest {
             assert(recyclerView.adapter?.itemCount == 1)
         }
     }
+
+    @Test
+    fun displayFragment_sharingPhotoWorks() {
+        runBlocking {
+            val imagePath = "test_image1.jpg"
+            uploadImageToStorageEmulator(context, imagePath, "test_image1.jpg")
+        }
+
+        Thread.sleep(WAIT_TIME_INIT)
+
+        onView(withId(R.id.refreshButton)).perform(click())
+
+        Thread.sleep(WAIT_TIME_REFRESH)
+
+        onView(withId(R.id.shareButton)).perform(click())
```

Comment:
If possible the use of thread.sleep should be avoided. Is there a possibility to wait until a new image appears instead ?

Reply:
Thread.sleep should be avoided in main code, it's not a issue in test code ^^

Reply:
I see your point and I agree with you. Most importantly test should be testing if a functionnability is working (which it is the case here). Therefore I can agree to keep this Thread.sleep as having a functionnality tested is more valuable than change the test at this point.